week_3/2018202184DHX/src/Crawler.cpp

```diff
@@ -0,0 +1,133 @@
+#include "Crawler.h"
+#include <iostream>
+#include <cstdio>
+#include <fstream>
+#include <string>
+#include <queue>
+#include <set>
+#include <cstdlib>
+#include <algorithm>
+#include <regex>
+
+int Crawler::count = 0;
+
+Crawler::Crawler() {
+    root = "http://econ.ruc.edu.cn/";   
+    save_path = "./";		
+}
+
+Crawler::Crawler(const std::string url) {
+	root = url;
+	save_path = "./";
+}
+
+Crawler::Crawler(const std::string url, const std::string path) {
+	root = url;
+	save_path = path;
+}
+
+Crawler::~Crawler() {
+
+}
+
+std::string Crawler::GetPageToFile(const std::string url, const std::string path,int count) {
+    //get system instruction to use wget
+    std::string file_name = std::to_string(count);
+    file_name += ".html";
+    std::string instruction_name = "wget -O " + path + file_name + " \"" + url + "\"";
+    const char *key = instruction_name.c_str();
+    system(key);
+    return file_name;  
+}
+
+void Crawler::ReadFile(const std::string file_name, const std::string path, std::string &content) {
+    //open file
+    std::ifstream file_to_be_read;
+    file_to_be_read.open(path + file_name);		
+
+    std::string temp;
+    while(getline(file_to_be_read, temp))
+        content += temp;
+    //close file
+    file_to_be_read.close();
+
+	return;
+}
+
+void Crawler::ParseOnePage(std::string &current_url, std::string &content, std::queue<std::string> &url_to_be_crawled) {
+        const std::string BlackList[100] = {
+    "@", "google", "csdn.", "pku", "ccf.org", "rss.", "math.ruc", "challenger.", "vedio_detail", "str = str+",
+    ".doc", ".ppt", ".PPT", ".zip", ".rar", ".xls", ".pdf", ".css", ".txt", ".jpg", ".JPG", ".png", "bmp", ".jpeg", 
+    ".js", ".flv", ".mp4", ".avi", "words_end"
+	};
+
+    std::regex pattern("(href=|HREF=)((\"(.*?)\")|(\'(.*?)\')|(([^\"\']*?)>))");

```

Comment:
`<a href=this://fucking/tag?breaks="your"#regex>`

`<A hReF=tHiS://fUcKiNg.TaG/aLsO?bReAkS=yOuR#rEgEx>`